java/arcs/core/data/Recipe.kt

```diff
@@ -40,9 +40,9 @@ data class Recipe(
         val type: Type,
         val storageKey: String? = null,
         val capabilities: Capabilities? = null,
-        val associatedHandles: List<String> = emptyList()
+        val associatedHandleNames: List<String> = emptyList(),
```

Comment:
I understand why this is comfortable when decoding, by ideally in the resulting data structure you would not have both associatedHandleNames and associatedHandles, can we explore ways to have only associatedHandles ?